clippy_lints/src/init_numbered_fields.rs

```diff
@@ -46,7 +45,7 @@ impl<'tcx> LateLintPass<'tcx> for NumberedFields {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, e: &'tcx Expr<'_>) {
         if let ExprKind::Struct(path, fields, None) = e.kind {
             if !fields.is_empty()
-                && !in_macro(e.span)
+                && !e.span.from_expansion()
```

Comment:
This is not completely behavior-preserving? What if the expansion was a desugaring?

Reply:
Right. If it is desugaring then it doesn't make sense to continue because the user can't change it.

Reply:
I see. Ok then.